pkg/plugin/datasource.go

```diff
@@ -228,7 +228,8 @@ func (d *Datasource) query(ctx context.Context, _ backend.PluginContext, query b
 	}
 
 	q.TimeRange = TimeRange(query.TimeRange)
-
+	q.MaxDataPoints = query.MaxDataPoints
```

Comment:
Would it be better to parse `query backend.DataQuery`  before this function call on L:131? It would be more clear and testable, I think.

Reply:
yes. Made this change